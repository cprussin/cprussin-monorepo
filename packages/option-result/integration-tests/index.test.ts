import { exec } from "node:child_process";
import { rm } from "node:fs/promises";
import path from "node:path";
import { promisify } from "node:util";

const ONE_SECOND_IN_MS = 1000;

const TEST_DIR = path.join(import.meta.dirname, "__fixtures__", "test-package");
const NODE_MODULES = path.join(TEST_DIR, "node_modules");

const execAsync = promisify(exec);

describe("integration", () => {
  beforeEach(async () => {
    await rm(NODE_MODULES, { recursive: true, force: true });
  });

  afterEach(async () => {
    await rm(NODE_MODULES, { recursive: true });
  });

  it(
    "installs & works",
    async () => {
      await execAsync("pnpm i", { cwd: TEST_DIR });
      const { stdout } = await execAsync("node ./index.js", { cwd: TEST_DIR });
      expect(stdout).toMatchSnapshot();
    },
    30 * ONE_SECOND_IN_MS,
  );
});
