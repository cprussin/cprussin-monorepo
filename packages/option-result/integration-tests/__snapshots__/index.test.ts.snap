// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration installs & works 1`] = `
"hello
world
"
`;
